### ISSUE TITLE
problems with huge addressbooks

### ISSUE THREAD

**Issue by markum** [nextcloud/contacts#176]
I have a rather huge addressbook with more than 1500 contacts, many of them containing pictures of the contacts. Unfortunately this results in issues when syncing with Android or the KDE addressbook. Not all contacts end up in the addressbook on Android and KDE. For Android I use Carddav, I have also used Davdroid. The result does not differ much. The best result I have archived by increasing the sockettimeout in the Carddav-App to 9999. I have also recently switch from my.owndrive.com as Owncloud-provider now to Manitu with a Nextcloud-installation. Here I initially also had the problem with uploading the addressbook. It never worked, I had to import it into Android and sync it too Nextcloud.

I know 1500 contacts is rather huge addressbook, but I believe there are other power users too. So I would expect wish that the contact app in Nextcloud should be able to handle big addressbooks in a better (faster) way so that syncing, uploading, retrieving hugs addressbooks works better. When logging into Nextcloud the contacts app also takes an eternity until the first contact is shown. So the problem also appear internally.

I am using Nextcloud 10.02 I do not know anything about server configuration, since I do not have SSH-access and installed it via one-click-install provided by Manitu.

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
Results: No errors have been found.

**List of activated apps:**
Activity 2.4.1
Collaborative tags 1.1.3
Comments 1.1.0
Deleted files 1.1.0
External storage support 1.1.2
Federation 1.1.1
File sharing 1.1.1
Files automated tagging 1.1.1
First run wizard 2.0
Gallery 16.0.0
Log Reader 2.0.0
Nextcloud announcements 1.0
Notifications 1.0.1
Password policy 1.1.0
PDF viewer 1.0.1
Server info 1.1.1
Share by mail 1.0.1
Text editor 2.2
Theming 1.1.1
Update notification 1.1.1
Versions 1.4.0
Video player 1.0.0
Activities for shared file downloads 1.0.1
Admin notifications 1.0.0
Announcement Center 3.0.0
AppOrder 0.3.3
Audio Player 1.4.1
Bookmarks 0.9.1
Calendar 1.5.1
Collabora Online 1.1.25
Contacts 1.5.3
Deck 0.1.1
Direct Menu 0.10.0
Issue Template 0.2.1
Nextant 1.0.6
Notizen 2.2.0
OwnBackup 17.3.0
Phone Sync 1.11.5
Reader (ebook reader) 1.0.1
Spreed.ME 0.3.8
Tasks 0.9.5
Video calls 1.2.0
Weather 1.3.5
**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

**Comment by Bullnados** [comment]
Perhaps u should update to 10.0.3 or 11.0.2 just for security

**Comment by irgendwie** [comment]
@markum If I understand you correctly (and please correct me if I am wrong) there are multiple problems at different points:
- First of all, you have problems with syncing contacts (apart from the contacts app), meaning external applications don't get all contacts?
- Second, the contacts app itself acts slow/unusable with that many contacts?

For the first problem, the CardDAV backend in nextcloud/server could be the cause since all contacts are exchanged via this backend. I assume your applications cause timeouts on the server/client side and there is no real way around it but to increase them. 😒
I can imagine why uploading contacts from Android works relatively good, since I assume one contact at a time is uploaded (due to network limitations).

For the second problem, the Contacts app (a 'simple' carddav client) currently is really bad at handling huge amounts of contacts, since it fetches all contacts (with all details) on page load.
This will be fixed in a future version of the Contacts app (ref. https://github.com/nextcloud/contacts/pull/88).
Additionally this new feature will probably only be available for nextcloud 11 upwards, since it requires changes in the CardDAV backend.

**Comment by myzinsky** [comment]
I have the same issues here: I have an address book with 400 contacts ...

- The web app does not show all the contacts
- The syncing to the iPhone syncs only half of all contacts

Any help is much appreciated!

**Comment by myzinsky** [comment]
An interesting fact: If I remove all the pictures from my vcard files and import this, everything works!

**Comment by skjnldsv** [comment]
Yes, big base64 encoded pictures aren't exactly browser friendly. It is pretty heavy.
Can we close of this issue is fixed?

**Comment by mokkin** [comment]
I also have the problem with the spinning wheel on my addressbooks. I upgraded from ownCloud 10 to NextCloud 12.0.1 a couple of days ago, but the problem didn't disappear.
Please check my description in the ownCloud ticket for further information:
https://github.com/owncloud/contacts/issues/430

**Comment by skjnldsv** [comment]
@markum can you try the current nightly build?

**Comment by skjnldsv** [comment]
Should be fixed with 3.0.0
